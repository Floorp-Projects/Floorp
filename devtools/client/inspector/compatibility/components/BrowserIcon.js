/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { PureComponent } = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const ICONS = {
  firefox: {
    src: "chrome://devtools/skin/images/browsers/firefox.svg",
    isMobileIconNeeded: false,
  },
  firefox_android: {
    src: "chrome://devtools/skin/images/browsers/firefox.svg",
    isMobileIconNeeded: true,
  },
  chrome: {
    src: "chrome://devtools/skin/images/browsers/chrome.svg",
    isMobileIconNeeded: false,
  },
  chrome_android: {
    src: "chrome://devtools/skin/images/browsers/chrome.svg",
    isMobileIconNeeded: true,
  },
  safari: {
    src: "chrome://devtools/skin/images/browsers/safari.svg",
    isMobileIconNeeded: false,
  },
  safari_ios: {
    src: "chrome://devtools/skin/images/browsers/safari.svg",
    isMobileIconNeeded: true,
  },
  edge: {
    src: "chrome://devtools/skin/images/browsers/edge.svg",
    isMobileIconNeeded: false,
  },
};

class BrowserIcon extends PureComponent {
  static get propTypes() {
    return {
      id: Types.browser.id,
      title: PropTypes.string.isRequired,
    };
  }

  render() {
    const { id, title } = this.props;

    const icon = ICONS[id];

    return dom.span(
      {
        className:
          "compatibility-browser-icon" +
          (icon.isMobileIconNeeded
            ? " compatibility-browser-icon--mobile"
            : ""),
      },
      dom.img({
        className: "compatibility-browser-icon__image",
        alt: title,
        title,
        src: icon.src,
      })
    );
  }
}

module.exports = BrowserIcon;
