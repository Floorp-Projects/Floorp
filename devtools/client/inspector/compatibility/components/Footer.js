/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

const {
  updateSettingsVisibility,
} = require("devtools/client/inspector/compatibility/actions/compatibility");

loader.lazyRequireGetter(
  this,
  "openDocLink",
  "devtools/client/shared/link",
  true
);

const FEEDBACK_LINK =
  "https://docs.google.com/forms/d/e/1FAIpQLSeevOHveQ1tDuKYY5Fxyb3vvbKKumdLWUT5-RuwJWoAAOST5g/viewform";

const REPORT_ICON = "chrome://devtools/skin/images/report.svg";
const SETTINGS_ICON = "chrome://devtools/skin/images/settings.svg";

class Footer extends PureComponent {
  static get propTypes() {
    return {
      updateSettingsVisibility: PropTypes.func.isRequired,
    };
  }

  _renderButton(icon, labelId, titleId, onClick) {
    return Localized(
      {
        id: titleId,
        attrs: { title: true },
      },
      dom.button(
        {
          className: "compatibility-footer__button",
          title: titleId,
          onClick,
        },
        dom.img({
          className: "compatibility-footer__icon",
          src: icon,
        }),
        Localized(
          {
            id: labelId,
          },
          dom.label(
            {
              className: "compatibility-footer__label",
            },
            labelId
          )
        )
      )
    );
  }

  render() {
    return dom.footer(
      {
        className: "compatibility-footer",
      },
      this._renderButton(
        SETTINGS_ICON,
        "compatibility-settings-button-label",
        "compatibility-settings-button-title",
        this.props.updateSettingsVisibility
      ),
      this._renderButton(
        REPORT_ICON,
        "compatibility-feedback-button-label",
        "compatibility-feedback-button-title",
        () => openDocLink(FEEDBACK_LINK)
      )
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateSettingsVisibility: () => dispatch(updateSettingsVisibility(true)),
  };
};

module.exports = connect(null, mapDispatchToProps)(Footer);
