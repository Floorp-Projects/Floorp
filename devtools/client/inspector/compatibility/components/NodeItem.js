/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { PureComponent } = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const {
  translateNodeFrontToGrip,
} = require("devtools/client/inspector/shared/utils");
const { REPS, MODE } = require("devtools/client/shared/components/reps/index");
const { Rep } = REPS;
const ElementNode = REPS.ElementNode;

const Types = require("devtools/client/inspector/compatibility/types");

const {
  highlightNode,
  unhighlightNode,
} = require("devtools/client/inspector/boxmodel/actions/box-model-highlighter");

class NodeItem extends PureComponent {
  static get propTypes() {
    return {
      dispatch: PropTypes.func.isRequired,
      node: Types.node.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const { dispatch, node, setSelectedNode } = this.props;

    return dom.li(
      { className: "compatibility-node-item" },
      Rep({
        defaultRep: ElementNode,
        mode: MODE.TINY,
        object: translateNodeFrontToGrip(node),
        onDOMNodeClick: () => {
          setSelectedNode(node);
          dispatch(unhighlightNode());
        },
        onDOMNodeMouseOut: () => dispatch(unhighlightNode()),
        onDOMNodeMouseOver: () => dispatch(highlightNode(node)),
      })
    );
  }
}

module.exports = NodeItem;
