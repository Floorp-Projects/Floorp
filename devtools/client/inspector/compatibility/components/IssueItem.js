/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const Services = require("Services");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

loader.lazyRequireGetter(
  this,
  "openDocLink",
  "devtools/client/shared/link",
  true
);

const UnsupportedBrowserList = createFactory(
  require("devtools/client/inspector/compatibility/components/UnsupportedBrowserList")
);

const Types = require("devtools/client/inspector/compatibility/types");

const NodePane = createFactory(
  require("devtools/client/inspector/compatibility/components/NodePane")
);

// For test
loader.lazyRequireGetter(
  this,
  "toSnakeCase",
  "devtools/client/inspector/compatibility/utils/cases",
  true
);

class IssueItem extends PureComponent {
  static get propTypes() {
    return {
      ...Types.issue,
      dispatch: PropTypes.func.isRequired,
      setSelectedNode: PropTypes.func.isRequired,
    };
  }

  constructor(props) {
    super(props);
    this._onLinkClicked = this._onLinkClicked.bind(this);
  }

  _onLinkClicked(e) {
    const { url } = this.props;

    e.preventDefault();
    e.stopPropagation();

    openDocLink(
      url +
        "?utm_source=devtools&utm_medium=inspector-compatibility&utm_campaign=default"
    );
  }

  _getTestDataAttributes() {
    const testDataSet = {};

    if (Services.prefs.getBoolPref("devtools.testing", false)) {
      for (const [key, value] of Object.entries(this.props)) {
        if (key === "nodes") {
          continue;
        }
        const datasetKey = `data-qa-${toSnakeCase(key)}`;
        testDataSet[datasetKey] = JSON.stringify(value);
      }
    }

    return testDataSet;
  }

  _renderAliases() {
    const { property } = this.props;
    let { aliases } = this.props;

    if (!aliases) {
      return null;
    }

    aliases = aliases.filter(alias => alias !== property);

    if (aliases.length === 0) {
      return null;
    }

    return dom.ul(
      {
        className: "compatibility-issue-item__aliases",
      },
      aliases.map(alias =>
        dom.li(
          {
            key: alias,
            className: "compatibility-issue-item__alias",
          },
          alias
        )
      )
    );
  }

  _renderCauses() {
    const { deprecated, experimental, prefixNeeded } = this.props;

    if (!deprecated && !experimental && !prefixNeeded) {
      return null;
    }

    let localizationId = "";

    if (deprecated && experimental && prefixNeeded) {
      localizationId =
        "compatibility-issue-deprecated-experimental-prefixneeded";
    } else if (deprecated && experimental) {
      localizationId = "compatibility-issue-deprecated-experimental";
    } else if (deprecated && prefixNeeded) {
      localizationId = "compatibility-issue-deprecated-prefixneeded";
    } else if (experimental && prefixNeeded) {
      localizationId = "compatibility-issue-experimental-prefixneeded";
    } else if (deprecated) {
      localizationId = "compatibility-issue-deprecated";
    } else if (experimental) {
      localizationId = "compatibility-issue-experimental";
    } else if (prefixNeeded) {
      localizationId = "compatibility-issue-prefixneeded";
    }

    return Localized(
      {
        id: localizationId,
      },
      dom.span(
        { className: "compatibility-issue-item__causes" },
        localizationId
      )
    );
  }

  _renderDescription() {
    const { property, url } = this.props;

    return dom.div(
      {
        className: "compatibility-issue-item__description",
      },
      dom.a(
        {
          className: "compatibility-issue-item__mdn-link devtools-monospace",
          href: url,
          title: url,
          onClick: e => this._onLinkClicked(e),
        },
        property
      ),
      this._renderCauses(),
      this._renderUnsupportedBrowserList()
    );
  }

  _renderNodeList() {
    const { dispatch, nodes, setSelectedNode } = this.props;

    if (!nodes) {
      return null;
    }

    return NodePane({
      dispatch,
      nodes,
      setSelectedNode,
    });
  }

  _renderUnsupportedBrowserList() {
    const { unsupportedBrowsers } = this.props;

    return unsupportedBrowsers.length
      ? UnsupportedBrowserList({ browsers: unsupportedBrowsers })
      : null;
  }

  render() {
    const {
      deprecated,
      experimental,
      property,
      unsupportedBrowsers,
    } = this.props;

    const classes = ["compatibility-issue-item"];

    if (deprecated) {
      classes.push("compatibility-issue-item--deprecated");
    }

    if (experimental) {
      classes.push("compatibility-issue-item--experimental");
    }

    if (unsupportedBrowsers.length) {
      classes.push("compatibility-issue-item--unsupported");
    }

    return dom.li(
      {
        className: classes.join(" "),
        key: property,
        ...this._getTestDataAttributes(),
      },
      this._renderDescription(),
      this._renderAliases(),
      this._renderNodeList()
    );
  }
}

module.exports = IssueItem;
