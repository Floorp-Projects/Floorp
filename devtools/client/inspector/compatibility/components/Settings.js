/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

const Types = require("devtools/client/inspector/compatibility/types");

const BrowserIcon = createFactory(
  require("devtools/client/inspector/compatibility/components/BrowserIcon")
);

const {
  updateSettingsVisibility,
  updateTargetBrowsers,
} = require("devtools/client/inspector/compatibility/actions/compatibility");

const CLOSE_ICON = "chrome://devtools/skin/images/close.svg";

class Settings extends PureComponent {
  static get propTypes() {
    return {
      defaultTargetBrowsers: PropTypes.arrayOf(PropTypes.shape(Types.browser))
        .isRequired,
      targetBrowsers: PropTypes.arrayOf(PropTypes.shape(Types.browser))
        .isRequired,
      updateTargetBrowsers: PropTypes.func.isRequired,
      updateSettingsVisibility: PropTypes.func.isRequired,
    };
  }

  constructor(props) {
    super(props);

    this._onTargetBrowserChanged = this._onTargetBrowserChanged.bind(this);

    this.state = {
      targetBrowsers: props.targetBrowsers,
    };
  }

  _onTargetBrowserChanged({ target }) {
    const { id, status } = target.dataset;
    let { targetBrowsers } = this.state;

    if (target.checked) {
      targetBrowsers = [...targetBrowsers, { id, status }];
    } else {
      targetBrowsers = targetBrowsers.filter(
        b => !(b.id === id && b.status === status)
      );
    }

    this.setState({ targetBrowsers });
  }

  _renderTargetBrowsers() {
    const { defaultTargetBrowsers } = this.props;
    const { targetBrowsers } = this.state;

    return dom.section(
      {
        className: "compatibility-settings__target-browsers",
      },
      Localized(
        { id: "compatibility-target-browsers-header" },
        dom.header(
          {
            className: "compatibility-settings__target-browsers-header",
          },
          "compatibility-target-browsers-header"
        )
      ),
      dom.ul(
        {
          className: "compatibility-settings__target-browsers-list",
        },
        defaultTargetBrowsers.map(({ id, name, status, version }) => {
          const inputId = `${id}-${status}`;
          const isTargetBrowser = !!targetBrowsers.find(
            b => b.id === id && b.status === status
          );
          return dom.li(
            {
              className: "compatibility-settings__target-browsers-item",
            },
            dom.input({
              id: inputId,
              type: "checkbox",
              checked: isTargetBrowser,
              onChange: this._onTargetBrowserChanged,
              "data-id": id,
              "data-status": status,
            }),
            dom.label(
              {
                className: "compatibility-settings__target-browsers-item-label",
                htmlFor: inputId,
              },
              BrowserIcon({ id, title: `${name} ${status}` }),
              `${name} ${status} (${version})`
            )
          );
        })
      )
    );
  }

  _renderHeader() {
    return dom.header(
      {
        className: "compatibility-settings__header",
      },
      Localized(
        { id: "compatibility-settings-header" },
        dom.label(
          {
            className: "compatibility-settings__header-label",
          },
          "compatibility-settings-header"
        )
      ),
      Localized(
        {
          id: "compatibility-close-settings-button",
          attrs: { title: true },
        },
        dom.button(
          {
            className: "compatibility-settings__header-button",
            title: "compatibility-close-settings-button",
            onClick: () => {
              const { defaultTargetBrowsers } = this.props;
              const { targetBrowsers } = this.state;

              // Sort by ordering of default browsers.
              const browsers = defaultTargetBrowsers.filter(b =>
                targetBrowsers.find(t => t.id === b.id && t.status === b.status)
              );

              if (
                this.props.targetBrowsers.toString() !== browsers.toString()
              ) {
                this.props.updateTargetBrowsers(browsers);
              }

              this.props.updateSettingsVisibility();
            },
          },
          dom.img({
            className: "compatibility-settings__header-icon",
            src: CLOSE_ICON,
          })
        )
      )
    );
  }

  render() {
    return dom.section(
      {
        className: "compatibility-settings",
      },
      this._renderHeader(),
      this._renderTargetBrowsers()
    );
  }
}

const mapStateToProps = state => {
  return {
    defaultTargetBrowsers: state.compatibility.defaultTargetBrowsers,
    targetBrowsers: state.compatibility.targetBrowsers,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateTargetBrowsers: browsers => dispatch(updateTargetBrowsers(browsers)),
    updateSettingsVisibility: () => dispatch(updateSettingsVisibility(false)),
  };
};

module.exports = connect(mapStateToProps, mapDispatchToProps)(Settings);
