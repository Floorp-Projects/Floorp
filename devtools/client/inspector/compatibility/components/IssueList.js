/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const Types = require("devtools/client/inspector/compatibility/types");

const IssueItem = createFactory(
  require("devtools/client/inspector/compatibility/components/IssueItem")
);

class IssueList extends PureComponent {
  static get propTypes() {
    return {
      dispatch: PropTypes.func.isRequired,
      issues: PropTypes.arrayOf(PropTypes.shape(Types.issue)).isRequired,
      setSelectedNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const { dispatch, issues, setSelectedNode } = this.props;

    return dom.ul(
      { className: "compatibility-issue-list" },
      issues.map(issue =>
        IssueItem({
          ...issue,
          dispatch,
          setSelectedNode,
        })
      )
    );
  }
}

module.exports = IssueList;
