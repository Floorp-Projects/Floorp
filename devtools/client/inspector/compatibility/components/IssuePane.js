/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");
const Localized = createFactory(FluentReact.Localized);

const Types = require("devtools/client/inspector/compatibility/types");

const IssueList = createFactory(
  require("devtools/client/inspector/compatibility/components/IssueList")
);

class IssuePane extends PureComponent {
  static get propTypes() {
    return {
      dispatch: PropTypes.func.isRequired,
      issues: PropTypes.arrayOf(PropTypes.shape(Types.issue)).isRequired,
      setSelectedNode: PropTypes.func.isRequired,
    };
  }

  _renderNoIssues() {
    return Localized(
      { id: "compatibility-no-issues-found" },
      dom.p(
        { className: "devtools-sidepanel-no-result" },
        "compatibility-no-issues-found"
      )
    );
  }

  render() {
    const { dispatch, issues, setSelectedNode } = this.props;

    return issues.length
      ? IssueList({
          dispatch,
          issues,
          setSelectedNode,
        })
      : this._renderNoIssues();
  }
}

module.exports = IssuePane;
