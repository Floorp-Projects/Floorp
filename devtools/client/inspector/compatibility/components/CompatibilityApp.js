/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

const { connect } = require("devtools/client/shared/vendor/react-redux");
const {
  createFactory,
  PureComponent,
} = require("devtools/client/shared/vendor/react");
const dom = require("devtools/client/shared/vendor/react-dom-factories");
const PropTypes = require("devtools/client/shared/vendor/react-prop-types");

const FluentReact = require("devtools/client/shared/vendor/fluent-react");

const Types = require("devtools/client/inspector/compatibility/types");

const Accordion = createFactory(
  require("devtools/client/shared/components/Accordion")
);
const Footer = createFactory(
  require("devtools/client/inspector/compatibility/components/Footer")
);
const IssuePane = createFactory(
  require("devtools/client/inspector/compatibility/components/IssuePane")
);
const Settings = createFactory(
  require("devtools/client/inspector/compatibility/components/Settings")
);

class CompatibilityApp extends PureComponent {
  static get propTypes() {
    return {
      dispatch: PropTypes.func.isRequired,
      // getString prop is injected by the withLocalization wrapper
      getString: PropTypes.func.isRequired,
      isSettingsVisibile: PropTypes.bool.isRequired,
      isTopLevelTargetProcessing: PropTypes.bool.isRequired,
      selectedNodeIssues: PropTypes.arrayOf(PropTypes.shape(Types.issue))
        .isRequired,
      topLevelTargetIssues: PropTypes.arrayOf(PropTypes.shape(Types.issue))
        .isRequired,
      setSelectedNode: PropTypes.func.isRequired,
    };
  }

  render() {
    const {
      dispatch,
      getString,
      isSettingsVisibile,
      isTopLevelTargetProcessing,
      selectedNodeIssues,
      topLevelTargetIssues,
      setSelectedNode,
    } = this.props;

    const selectedNodeIssuePane = IssuePane({
      issues: selectedNodeIssues,
    });

    const topLevelTargetIssuePane =
      topLevelTargetIssues.length > 0 || !isTopLevelTargetProcessing
        ? IssuePane({
            dispatch,
            issues: topLevelTargetIssues,
            setSelectedNode,
          })
        : null;

    const throbber = isTopLevelTargetProcessing
      ? dom.div({
          className: "compatibility-app__throbber devtools-throbber",
        })
      : null;

    return dom.section(
      {
        className: "compatibility-app theme-sidebar inspector-tabpanel",
      },
      dom.div(
        {
          className:
            "compatibility-app__container" +
            (isSettingsVisibile ? " compatibility-app__container-hidden" : ""),
        },
        Accordion({
          className: "compatibility-app__main",
          items: [
            {
              id: "compatibility-app--selected-element-pane",
              header: getString("compatibility-selected-element-header"),
              component: selectedNodeIssuePane,
              opened: true,
            },
            {
              id: "compatibility-app--all-elements-pane",
              header: getString("compatibility-all-elements-header"),
              component: [topLevelTargetIssuePane, throbber],
              opened: true,
            },
          ],
        }),
        Footer({
          className: "compatibility-app__footer",
        })
      ),
      isSettingsVisibile ? Settings() : null
    );
  }
}

const mapStateToProps = state => {
  return {
    isSettingsVisibile: state.compatibility.isSettingsVisibile,
    isTopLevelTargetProcessing: state.compatibility.isTopLevelTargetProcessing,
    selectedNodeIssues: state.compatibility.selectedNodeIssues,
    topLevelTargetIssues: state.compatibility.topLevelTargetIssues,
  };
};
module.exports = FluentReact.withLocalization(
  connect(mapStateToProps)(CompatibilityApp)
);
