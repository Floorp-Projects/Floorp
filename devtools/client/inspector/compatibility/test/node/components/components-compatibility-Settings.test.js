/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

/**
 * Unit tests for the Settings component.
 */

const { shallow } = require("enzyme");
const { createFactory } = require("devtools/client/shared/vendor/react");
const { thunk } = require("devtools/client/shared/redux/middleware/thunk.js");
const configureStore = require("redux-mock-store").default;

const Settings = createFactory(
  require("devtools/client/inspector/compatibility/components/Settings")
);

const DEFAULT_BROWSERS = [
  { id: "firefox", name: "Firefox", status: "nightly", version: "78" },
  { id: "firefox", name: "Firefox", status: "beta", version: "77" },
  { id: "firefox", name: "Firefox", status: "current", version: "76" },
];

describe("Settings component", () => {
  it("renders default browsers with no selected browsers", () => {
    const mockStore = configureStore([thunk()]);
    const store = mockStore({
      compatibility: {
        defaultTargetBrowsers: DEFAULT_BROWSERS,
        targetBrowsers: [],
      },
    });

    const connectWrapper = shallow(Settings({ store }));
    const targetComponent = connectWrapper.dive();
    expect(targetComponent).toMatchSnapshot();
  });

  it("renders default browsers with a selected browsers", () => {
    const mockStore = configureStore([thunk()]);
    const store = mockStore({
      compatibility: {
        defaultTargetBrowsers: DEFAULT_BROWSERS,
        targetBrowsers: [DEFAULT_BROWSERS[1]],
      },
    });

    const connectWrapper = shallow(Settings({ store }));
    const targetComponent = connectWrapper.dive();
    expect(targetComponent).toMatchSnapshot();
  });

  it("renders default browsers with full selected browsers", () => {
    const mockStore = configureStore([thunk()]);
    const store = mockStore({
      compatibility: {
        defaultTargetBrowsers: DEFAULT_BROWSERS,
        targetBrowsers: DEFAULT_BROWSERS,
      },
    });

    const connectWrapper = shallow(Settings({ store }));
    const targetComponent = connectWrapper.dive();
    expect(targetComponent).toMatchSnapshot();
  });
});
