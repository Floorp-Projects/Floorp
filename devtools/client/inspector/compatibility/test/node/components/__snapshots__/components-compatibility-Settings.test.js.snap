// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders default browsers with a selected browsers 1`] = `
<section
  className="compatibility-settings"
>
  <header
    className="compatibility-settings__header"
  >
    <Localized
      id="compatibility-settings-header"
    >
      <label
        className="compatibility-settings__header-label"
      >
        compatibility-settings-header
      </label>
    </Localized>
    <Localized
      attrs={
        Object {
          "title": true,
        }
      }
      id="compatibility-close-settings-button"
    >
      <button
        className="compatibility-settings__header-button"
        onClick={[Function]}
        title="compatibility-close-settings-button"
      >
        <img
          className="compatibility-settings__header-icon"
          src="chrome://devtools/skin/images/close.svg"
        />
      </button>
    </Localized>
  </header>
  <section
    className="compatibility-settings__target-browsers"
  >
    <Localized
      id="compatibility-target-browsers-header"
    >
      <header
        className="compatibility-settings__target-browsers-header"
      >
        compatibility-target-browsers-header
      </header>
    </Localized>
    <ul
      className="compatibility-settings__target-browsers-list"
    >
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={false}
          data-id="firefox"
          data-status="nightly"
          id="firefox-nightly"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-nightly"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox nightly"
          />
          Firefox nightly (78)
        </label>
      </li>
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={true}
          data-id="firefox"
          data-status="beta"
          id="firefox-beta"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-beta"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox beta"
          />
          Firefox beta (77)
        </label>
      </li>
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={false}
          data-id="firefox"
          data-status="current"
          id="firefox-current"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-current"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox current"
          />
          Firefox current (76)
        </label>
      </li>
    </ul>
  </section>
</section>
`;

exports[`Settings component renders default browsers with full selected browsers 1`] = `
<section
  className="compatibility-settings"
>
  <header
    className="compatibility-settings__header"
  >
    <Localized
      id="compatibility-settings-header"
    >
      <label
        className="compatibility-settings__header-label"
      >
        compatibility-settings-header
      </label>
    </Localized>
    <Localized
      attrs={
        Object {
          "title": true,
        }
      }
      id="compatibility-close-settings-button"
    >
      <button
        className="compatibility-settings__header-button"
        onClick={[Function]}
        title="compatibility-close-settings-button"
      >
        <img
          className="compatibility-settings__header-icon"
          src="chrome://devtools/skin/images/close.svg"
        />
      </button>
    </Localized>
  </header>
  <section
    className="compatibility-settings__target-browsers"
  >
    <Localized
      id="compatibility-target-browsers-header"
    >
      <header
        className="compatibility-settings__target-browsers-header"
      >
        compatibility-target-browsers-header
      </header>
    </Localized>
    <ul
      className="compatibility-settings__target-browsers-list"
    >
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={true}
          data-id="firefox"
          data-status="nightly"
          id="firefox-nightly"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-nightly"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox nightly"
          />
          Firefox nightly (78)
        </label>
      </li>
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={true}
          data-id="firefox"
          data-status="beta"
          id="firefox-beta"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-beta"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox beta"
          />
          Firefox beta (77)
        </label>
      </li>
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={true}
          data-id="firefox"
          data-status="current"
          id="firefox-current"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-current"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox current"
          />
          Firefox current (76)
        </label>
      </li>
    </ul>
  </section>
</section>
`;

exports[`Settings component renders default browsers with no selected browsers 1`] = `
<section
  className="compatibility-settings"
>
  <header
    className="compatibility-settings__header"
  >
    <Localized
      id="compatibility-settings-header"
    >
      <label
        className="compatibility-settings__header-label"
      >
        compatibility-settings-header
      </label>
    </Localized>
    <Localized
      attrs={
        Object {
          "title": true,
        }
      }
      id="compatibility-close-settings-button"
    >
      <button
        className="compatibility-settings__header-button"
        onClick={[Function]}
        title="compatibility-close-settings-button"
      >
        <img
          className="compatibility-settings__header-icon"
          src="chrome://devtools/skin/images/close.svg"
        />
      </button>
    </Localized>
  </header>
  <section
    className="compatibility-settings__target-browsers"
  >
    <Localized
      id="compatibility-target-browsers-header"
    >
      <header
        className="compatibility-settings__target-browsers-header"
      >
        compatibility-target-browsers-header
      </header>
    </Localized>
    <ul
      className="compatibility-settings__target-browsers-list"
    >
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={false}
          data-id="firefox"
          data-status="nightly"
          id="firefox-nightly"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-nightly"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox nightly"
          />
          Firefox nightly (78)
        </label>
      </li>
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={false}
          data-id="firefox"
          data-status="beta"
          id="firefox-beta"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-beta"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox beta"
          />
          Firefox beta (77)
        </label>
      </li>
      <li
        className="compatibility-settings__target-browsers-item"
      >
        <input
          checked={false}
          data-id="firefox"
          data-status="current"
          id="firefox-current"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="compatibility-settings__target-browsers-item-label"
          htmlFor="firefox-current"
        >
          <BrowserIcon
            id="firefox"
            title="Firefox current"
          />
          Firefox current (76)
        </label>
      </li>
    </ul>
  </section>
</section>
`;
